### PR TITLE
Cleanup `Cask::Caskroom::casks`

### PR DESCRIPTION
**Pull request by Rylan12**
~This PR uses the new `/api/cask-source.json` endpoint to make loading casks using the API must faster. This is super noticeable in `brew outdated` when you have casks installed. Now, there's only one file to download that only needs to update whenever there's a new version available.~

~This PR depends on https://github.com/Homebrew/formulae.brew.sh/pull/654~

Cleanup `Cask::Caskroom::casks` so that it defaults to using a plain `Cask::CaskLoader::load` call. This will more reliably load the most up-to-date version of the cask without needing an extra `if Homebrew::EnvConfig.install_from_api?` call. `::load` uses nearly the same order to check for available casks to load:
1. Check API (if `HOMEBREW_INSTALL_FROM_API` is set)
2. Check `homebrew/cask`
3. Check other taps
4. Load from the caskfile in the metadata

This PR adds a new exception called `TapCaskAmbiguityException` that is raised when a cask exists in multiple taps (when neither is `homebrew/cask`). Now, `Cask::Caskroom::casks` will catch this exception and just choose the first available casks

### CODE REVIEW

File: Library/Homebrew/cask/caskroom.rb
Comment (Rylan12) [inline Library/Homebrew/cask/caskroom.rb:50]: Can someone from @Homebrew/cask confirm that this is an acceptable change?

As far as I can tell, `CaskLoader::load` will just try to load in the same order specified here. Is there another reason we needed special handling here?

**Comment by Rylan12** [comment]
Okay, I've removed the API part of this PR since, for now at least, we're going to do this a different way. However, I left the `Cask::Caskroom::casks` changes since I think those still make sense. However, I still would like to get a second opinion on these changes before merging.

File: Library/Homebrew/cask/caskroom.rb
Comment (SMillerDev) [reply Library/Homebrew/cask/caskroom.rb:50]: I have no idea, since I don't often modify cask code in brew. But it seems like these are legacy paths?

Comment (Rylan12) [reply Library/Homebrew/cask/caskroom.rb:50]: Yeah, that's kind of what I figured, but the bodies of the `if` and `elsif` statements are already handled in `CaskLoader::load` in the same order (the `tap_paths` is done [here](https://github.com/Homebrew/brew/blob/9416032ee5d253fc00419f111373af54e1c6bc5d/Library/Homebrew/cask/cask_loader.rb#L228-L240) and the metadata path is handled [here](https://github.com/Homebrew/brew/blob/9416032ee5d253fc00419f111373af54e1c6bc5d/Library/Homebrew/cask/cask_loader.rb#L246-L247)), so I can't figure out why `CaskLoader::load` would return anything different from what this is doing.

Comment (SMillerDev) [reply Library/Homebrew/cask/caskroom.rb:50]: Might just have been forgotten.

Comment (Rylan12) [reply Library/Homebrew/cask/caskroom.rb:50]: Actually, upon further inspection, there is one difference which is that `CaskLoader::load` will raise an error if the name token appears in multiple taps (aside from `homebrew/cask`) while this method will just choose the first tap which isn't necessarily the main tap.

I think it would make more sense to completely mimic `CaskLoader::load` except that when there is ambiguity, handle it better than just erroring.

**Review by MikeMcQuaid — APPROVED** [review]
Lovely!

**Pull request merged**
